src/transactions.rs

```diff
@@ -198,34 +220,108 @@ impl fmt::Display for FileContractID {
     }
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct FileContract {
+    #[serde(rename = "filesize")]
     pub file_size: u64,
     pub file_merkle_root: Hash256,
     pub window_start: u64,
     pub window_end: u64,
     pub payout: Currency,
     pub valid_proof_outputs: Vec<SiacoinOutput>,
     pub missed_proof_outputs: Vec<SiacoinOutput>,
-    pub unlock_hash: Address,
+    pub unlock_hash: Hash256,
     pub revision_number: u64,
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, PartialEq, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct FileContractRevision {
     #[serde(rename = "parentID")]
     pub parent_id: FileContractID,
     pub unlock_conditions: UnlockConditions,
     pub revision_number: u64,
+    #[serde(rename = "filesize")]
     pub file_size: u64,
     pub file_merkle_root: Hash256,
     pub window_start: u64,
     pub window_end: u64,
     pub valid_proof_outputs: Vec<SiacoinOutput>,
     pub missed_proof_outputs: Vec<SiacoinOutput>,
-    pub unlock_hash: Address,
+    pub unlock_hash: Hash256,
+}
+
+impl Serialize for FileContractRevision {
```

Comment:
So apparently something funky we do in `core` is serialize the `payout` field for `FileContractRevision`s when json serializing them. That makes it necessary to implement this.
If we decide to drop the `payout` field we can get rid of this again at the cost of not being compatible with `core` in that regard.

Reply:
hmm, I think we should remove that field from the `core` JSON serialization